src/Cache/QueryCache.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Db\Cache;
+
+use Psr\SimpleCache\CacheInterface;
+use Yiisoft\Cache\CacheKeyNormalizer;
+use Yiisoft\Cache\Dependency\Dependency;
+
+use function array_pop;
+use function end;
+use function is_array;
+
+final class QueryCache
+{
+    private CacheInterface $cache;
+    private bool $enableCache = true;
+    public array $cacheInfo = [];
+    private int $cacheDuration = 3600;
+    private CacheKeyNormalizer $cacheKeyNormalizer;
+
+    public function __construct(CacheInterface $cache, CacheKeyNormalizer $cacheKeyNormalizer)
+    {
+        $this->cache = $cache;
+        $this->cacheKeyNormalizer = $cacheKeyNormalizer;
+    }
+
+    public function normalize($key): string
+    {
+        return $this->cacheKeyNormalizer->normalize($key);
+    }
+
+    public function getCacheDuration(): ?int
+    {
+        return $this->cacheDuration;
+    }
+
+    public function getCacheInfo(): array
+    {
+        return $this->cacheInfo;
+    }
+
+    public function isCacheEnabled(): bool
```

Comment:
It doesn't seem to be used anywhere. Probably could be removed?

Reply:
It is good to leave it for any verification.